rafthttp/transport.go

```diff
@@ -37,6 +38,13 @@ type Raft interface {
 	ReportSnapshot(id uint64, status raft.SnapshotStatus)
 }
 
+type SnapshotStore interface {
```

Comment:
will add.

I originally mentioned to avoid snapshot concept is defined in rafthttp package, which is totally weird. Now it turns out that the interface does not need to understand what the snapshot is anymore, and it just provides io.Reader and io.Writer to read/write snapshot. So it is fine to put the interface here.